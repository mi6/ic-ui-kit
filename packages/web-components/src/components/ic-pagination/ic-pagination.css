@import "../../global/normalize.css";

:host {
  box-sizing: border-box;
  width: 100%;
  display: flex;
  justify-content: center;
}

nav {
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-start;
  align-items: center;
}

ic-button {
  cursor: pointer;
}

ic-button.next-previous {
  --icon-width: var(--ic-space-xs);
  --icon-height: calc(var(--ic-space-xs) + var(--ic-space-xxxs));

  padding: 0 var(--ic-space-xxs) 0 var(--ic-space-xxxs);
}

ic-button.first-last {
  --icon-width: calc(var(--ic-space-sm) + var(--ic-space-xxxs));
  --icon-height: calc(var(--ic-space-xs) + var(--ic-space-xxxs));
}

.disabled {
  color: var(--ic-architectural-200);
  pointer-events: none;
  cursor: none;
  user-select: none;
}

.hide-current-page {
  display: none;
}
