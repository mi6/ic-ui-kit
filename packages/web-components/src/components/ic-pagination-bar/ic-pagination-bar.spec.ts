import { PaginationBar } from "./ic-pagination-bar";
import { Pagination } from "../ic-pagination/ic-pagination";
import { Button } from "../ic-button/ic-button";
import { Tooltip } from "../ic-tooltip/ic-tooltip";
import { TextField } from "../ic-text-field/ic-text-field";
import { DEVICE_SIZES } from "../../utils/helpers";
import { waitForTimeout } from "../../testspec.setup";
import { newSpecPage } from "@stencil/core/testing";

describe("ic-pagination-bar", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with items per page controls", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with go to page controls", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with data pagination type", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" pagination-type="data"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with complex pagination controls", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" pagination-controls="complex"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a custom page label", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" page-label="Sheets"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a custom item label", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" pagination-type="data" item-label="Rows"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with left alignment", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" alignment="left"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with space between alignment", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" alignment="space-between"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with light appearance", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" appearance="light"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with dark appearance", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" apperance="dark"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with custom items per page options", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    paginationBar.itemsPerPageOptions = [
      { label: "15", value: "15" },
      { label: "30", value: "30" },
      { label: "60", value: "60" },
      { label: "All", value: "100" },
    ];

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should only allow a maximum of 4 custom items per page options", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="150" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    paginationBar.itemsPerPageOptions = [
      { label: "25", value: "25" },
      { label: "50", value: "50" },
      { label: "75", value: "75" },
      { label: "100", value: "100" },
      { label: "All", value: "150" },
    ];

    page.rootInstance.trimItemsPerPageOptions();

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should remove items per page options larger than the maximum number of items", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    paginationBar.itemsPerPageOptions = [
      { label: "25", value: "25" },
      { label: "50", value: "50" },
      { label: "150", value: "150" },
      { label: "All", value: "100" },
    ];

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should wrap pagination when the device size is small", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    await page.rootInstance.runResizeObserver();

    await page.waitForChanges();

    await page.rootInstance.resizeObserverCallback(DEVICE_SIZES.S);

    await page.waitForChanges();

    await page.rootInstance.paginationShouldWrap();

    await page.waitForChanges();

    expect(page).toMatchSnapshot();

    await page.rootInstance.disconnectedCallback();
  });

  it("should adjust the total page count when items per page is changed", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination],
      html: `<ic-pagination-bar total-items="100" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const select = paginationBar.shadowRoot.querySelector("ic-select");

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    expect(text.textContent).toEqual("Page 1 of 10");

    select.value = "25";

    page.rootInstance.changeItemsPerPage();

    page.rootInstance.setNumberPages();

    await page.waitForChanges();

    expect(select.value).toEqual("25");

    expect(text.textContent).toEqual("Page 1 of 4");
  });

  it("should change page when the pagination controls are clicked", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Button],
      html: `<ic-pagination-bar total-items="100"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const pagination = paginationBar.shadowRoot.querySelector("ic-pagination");

    const nextButton = pagination.shadowRoot.querySelector(
      "#next-page-button"
    ) as HTMLElement;

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    expect(text.textContent).toEqual("Page 1 of 10");

    await nextButton.click();

    page.rootInstance.changePage(2);

    await page.waitForChanges();

    expect(text.textContent).toEqual("Page 2 of 10");
  });

  it("should change page when the go to page button is clicked", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const button = paginationBar.shadowRoot.querySelector("ic-button");

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    const currentPage = page.rootInstance.currentPage;

    expect(text.textContent).toEqual("Page 1 of 10");

    expect(currentPage).toEqual(1);

    input.value = "3";

    button.click();

    await page.waitForChanges();

    expect(text.textContent).toEqual("Page 3 of 10");
  });

  it("should change page when enter is pressed on the go to page input", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    const currentPage = page.rootInstance.currentPage;

    const event = new KeyboardEvent("keydown", { key: "Enter" });

    expect(text.textContent).toEqual("Page 1 of 10");

    expect(currentPage).toEqual(1);

    input.value = "3";

    input.focus();

    await page.rootInstance.handleKeydown(event);

    await waitForTimeout(100);

    await page.waitForChanges();

    expect(text.textContent).toEqual("Page 3 of 10");
  });

  it("should error immediately when an invalid page is entered before it is submitted", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    expect(input.validationStatus).toBeUndefined();

    input.value = "15";

    input.focus();

    await page.rootInstance.handleInputChange();

    await page.waitForChanges();

    expect(input.validationStatus).toEqual("error");
  });

  it("should remain in error state if enter is pressed while in error state", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const event = new KeyboardEvent("keydown", { key: "Enter" });

    expect(input.validationStatus).toBeUndefined();

    input.value = "15";

    input.focus();

    await page.rootInstance.handleInputChange();

    await page.waitForChanges();

    expect(input.validationStatus).toEqual("error");

    page.rootInstance.handleKeydown(event);

    await waitForTimeout(100);

    await page.waitForChanges();

    expect(input.validationStatus).toEqual("error");
  });

  it("should not change page when the go to page button is clicked with an invalid page", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, TextField],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const button = paginationBar.shadowRoot.querySelector("ic-button");

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    const currentPage = page.rootInstance.currentPage;

    expect(text.textContent).toEqual("Page 1 of 10");

    expect(currentPage).toEqual(1);

    input.value = "15";

    button.click();

    await page.waitForChanges();

    expect(text.textContent).toEqual("Page 1 of 10");
  });

  it("should not change page when enter is pressed on the go to page input with an invalid page", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip, TextField],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    const currentPage = page.rootInstance.currentPage;

    const event = new KeyboardEvent("keydown", { key: "Enter" });

    expect(text.textContent).toEqual("Page 1 of 10");

    expect(currentPage).toEqual(1);

    input.value = "15";

    input.focus();

    await page.rootInstance.handleKeydown(event);

    await waitForTimeout(100);

    await page.waitForChanges();

    expect(text.textContent).toEqual("Page 1 of 10");
  });

  it("should hide the tooltip when the input loses focus", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const tooltip = paginationBar.shadowRoot.querySelector("ic-tooltip");

    expect(tooltip.getAttribute("data-show")).toBeNull();

    input.focus();

    input.value = "15";

    tooltip.displayTooltip(true, false);

    await page.waitForChanges();

    input.blur();

    await page.waitForChanges();

    expect(tooltip.getAttribute("data-show")).toBeNull();
  });

  it("should remove the error status when the input loses focus if the value is not invalid", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    input.focus();

    input.value = "";

    input.validationStatus = "error";

    await page.waitForChanges();

    input.blur();

    await page.waitForChanges();

    expect(input.validationStatus).toBe("");
  });

  it("should show the tooltip when the input is focussed if it is in an error state", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    input.focus();

    input.value = "15";

    input.validationStatus = "error";

    await page.waitForChanges();

    input.blur();

    await page.waitForChanges();

    input.focus();

    await page.waitForChanges();

    expect(input.validationStatus).toBe("error");
  });

  it("should error when a non numeric value is entered", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const event = new KeyboardEvent("keydown", { key: "a" });

    expect(input.validationStatus).toBeUndefined();

    input.focus();

    await page.rootInstance.handleKeyUp(event);

    await page.waitForChanges();

    expect(input.validationStatus).toEqual("error");
  });

  it("should remove the non-numeric error when a number is entered", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const errorEvent = new KeyboardEvent("keydown", { key: "a" });

    const successEvent = new KeyboardEvent("keydown", { key: "5" });

    expect(input.validationStatus).toBeUndefined();

    input.focus();

    await page.rootInstance.handleKeyUp(errorEvent);

    await page.waitForChanges();

    expect(input.validationStatus).toEqual("error");

    await page.rootInstance.handleKeydown(successEvent);

    await waitForTimeout(100);

    await page.rootInstance.handleInputChange();

    await page.waitForChanges();

    expect(input.validationStatus).toEqual("");
  });

  it("should remove the error state when backspace is pressed", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Tooltip],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const event = new KeyboardEvent("keydown", { key: "Backspace" });

    expect(input.validationStatus).toBeUndefined();

    input.focus();

    input.value = "15";

    input.validationStatus = "error";

    await page.waitForChanges();

    expect(input.validationStatus).toEqual("error");

    await page.rootInstance.handleKeydown(event);

    await waitForTimeout(100);

    await page.waitForChanges();

    expect(input.validationStatus).toEqual("");
  });
});
