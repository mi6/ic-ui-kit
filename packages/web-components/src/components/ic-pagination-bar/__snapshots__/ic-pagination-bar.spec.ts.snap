// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-pagination-bar should only allow a maximum of 4 custom items per page options 1`] = `
<ic-pagination-bar show-items-per-page-control="true" total-items="150">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <div class="items-per-page-holder">
          <ic-typography class="items-per-page-control-label pagination-text-default" variant="label">
            Items per page
          </ic-typography>
          <ic-select class="items-per-page-input" hidelabel="" label="items-per-page-input" small="" value="25"></ic-select>
        </div>
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 6
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="6" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should remove items per page options larger than the maximum number of items 1`] = `
<ic-pagination-bar show-items-per-page-control="true" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <div class="items-per-page-holder">
          <ic-typography class="items-per-page-control-label pagination-text-default" variant="label">
            Items per page
          </ic-typography>
          <ic-select class="items-per-page-input" hidelabel="" label="items-per-page-input" small="" value="10"></ic-select>
        </div>
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render 1`] = `
<ic-pagination-bar total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with a custom item label 1`] = `
<ic-pagination-bar item-label="Rows" pagination-type="data" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="item-pagination-label pagination-text-default" variant="label">
          1 - 10 of 100 rowss
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with a custom page label 1`] = `
<ic-pagination-bar page-label="Sheets" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Sheets 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with complex pagination controls 1`] = `
<ic-pagination-bar pagination-controls="complex" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with custom items per page options 1`] = `
<ic-pagination-bar show-items-per-page-control="true" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <div class="items-per-page-holder">
          <ic-typography class="items-per-page-control-label pagination-text-default" variant="label">
            Items per page
          </ic-typography>
          <ic-select class="items-per-page-input" hidelabel="" label="items-per-page-input" small="" value="10"></ic-select>
        </div>
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with dark appearance 1`] = `
<ic-pagination-bar apperance="dark" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with data pagination type 1`] = `
<ic-pagination-bar pagination-type="data" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="item-pagination-label pagination-text-default" variant="label">
          1 - 10 of 100 items
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with go to page controls 1`] = `
<ic-pagination-bar show-go-to-page-control="true" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
        <div class="go-to-page-holder">
          <ic-typography class="pagination-text-default" variant="label">
            Go to page
          </ic-typography>
          <ic-tooltip disableclick="" disablehover="" label="Please enter a valid page" target="#go-to-page-input">
            <ic-text-field class="go-to-page-input" hidelabel="" id="go-to-page-input" label="go-to-page-input" max="10" min="1" small="" type="number" validationinlineinternal=""></ic-text-field>
          </ic-tooltip>
          <ic-button appearance="default" class="go-to-page-button" size="small" variant="secondary">
            Go
          </ic-button>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with items per page controls 1`] = `
<ic-pagination-bar show-items-per-page-control="true" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <div class="items-per-page-holder">
          <ic-typography class="items-per-page-control-label pagination-text-default" variant="label">
            Items per page
          </ic-typography>
          <ic-select class="items-per-page-input" hidelabel="" label="items-per-page-input" small="" value="10"></ic-select>
        </div>
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with left alignment 1`] = `
<ic-pagination-bar alignment="left" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-left">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with light appearance 1`] = `
<ic-pagination-bar appearance="light" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-light" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="light" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with space between alignment 1`] = `
<ic-pagination-bar alignment="space-between" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-space-between">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should wrap pagination when the device size is small 1`] = `
Object {
  "body": <body>
  <ic-pagination-bar show-go-to-page-control="true" total-items="100">
    <mock:shadow-root>
      <div class="pagination-bar pagination-bar-right">
        <div class="item-controls">
          <ic-typography class="page-pagination-label pagination-text-default" variant="label">
            Page 1 of 10
          </ic-typography>
        </div>
        <div class="pagination-controls">
          <div class="pagination-holder">
            <ic-pagination>
              <mock:shadow-root>
                <nav aria-label="Pagination Navigation" role="navigation">
                  <ic-button appearance="default" aria-label="Go to first page" class="first-last page-button" disabled="" disabletooltip="" id="first-page-button" variant="icon">
                    svg
                  </ic-button>
                  <ic-button appearance="default" aria-label="Go to previous page" class="flip next-previous page-button" disabled="" disabletooltip="" id="previous-page-button" variant="icon">
                    svg
                  </ic-button>
                  <ic-pagination-item appearance="default" label="Page" page="1" pages="10" type="simple-current"></ic-pagination-item>
                  <ic-button appearance="default" aria-label="Go to next page" class="next-previous page-button" disabletooltip="" id="next-page-button" variant="icon">
                    svg
                  </ic-button>
                  <ic-button appearance="default" aria-label="Go to last page" class="first-last flip page-button" disabletooltip="" id="last-page-button" variant="icon">
                    svg
                  </ic-button>
                </nav>
              </mock:shadow-root>
            </ic-pagination>
          </div>
          <div class="go-to-page-holder">
            <ic-typography class="pagination-text-default" variant="label">
              Go to page
            </ic-typography>
            <ic-tooltip disableclick="" disablehover="" label="Please enter a valid page" target="#go-to-page-input">
              <ic-text-field class="go-to-page-input" hidelabel="" id="go-to-page-input" label="go-to-page-input" max="10" min="1" small="" type="number" validationinlineinternal=""></ic-text-field>
            </ic-tooltip>
            <ic-button appearance="default" class="go-to-page-button" size="small" variant="secondary">
              Go
            </ic-button>
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ic-pagination-bar>
</body>,
  "build": Object {
    "allRenderFn": true,
    "appendChildSlotFix": false,
    "asyncLoading": true,
    "asyncQueue": true,
    "attachStyles": false,
    "cloneNodeFix": false,
    "cmpDidLoad": true,
    "cmpDidRender": true,
    "cmpDidUnload": true,
    "cmpDidUpdate": true,
    "cmpShouldUpdate": true,
    "cmpWillLoad": true,
    "cmpWillRender": true,
    "cmpWillUpdate": true,
    "connectedCallback": true,
    "constructableCSS": true,
    "cssAnnotations": false,
    "cssVarShim": true,
    "devTools": false,
    "disconnectedCallback": true,
    "dynamicImportShim": false,
    "element": true,
    "event": true,
    "hasRenderFn": true,
    "hostListener": true,
    "hostListenerTarget": true,
    "hostListenerTargetBody": true,
    "hostListenerTargetDocument": true,
    "hostListenerTargetParent": true,
    "hostListenerTargetWindow": true,
    "hotModuleReplacement": false,
    "hydrateClientSide": false,
    "hydrateServerSide": false,
    "hydratedAttribute": false,
    "hydratedClass": true,
    "initializeNextTick": true,
    "invisiblePrehydration": true,
    "isDebug": false,
    "isDev": true,
    "isTesting": true,
    "lazyLoad": true,
    "lifecycle": true,
    "lifecycleDOMEvents": true,
    "member": true,
    "method": true,
    "mode": true,
    "observeAttribute": true,
    "profile": true,
    "prop": true,
    "propBoolean": true,
    "propMutable": true,
    "propNumber": true,
    "propString": true,
    "reflect": true,
    "safari10": false,
    "scoped": true,
    "scopedSlotTextContentFix": false,
    "scriptDataOpts": false,
    "shadowDelegatesFocus": true,
    "shadowDom": true,
    "shadowDomShim": false,
    "slot": true,
    "slotChildNodesFix": false,
    "slotRelocation": true,
    "state": true,
    "style": true,
    "svg": true,
    "taskQueue": true,
    "transformTagName": true,
    "updatable": true,
    "vdomAttribute": true,
    "vdomClass": true,
    "vdomFunctional": true,
    "vdomKey": true,
    "vdomListener": true,
    "vdomPropOrAttr": true,
    "vdomRef": true,
    "vdomRender": true,
    "vdomStyle": true,
    "vdomText": true,
    "vdomXlink": true,
    "watchCallback": true,
  },
  "doc": ,
  "flushLoadModule": [Function],
  "flushQueue": [Function],
  "setContent": [Function],
  "styles": Map {},
  "waitForChanges": [Function],
  "win": [MockWindow],
}
`;
